cfme/middleware/__init__.py

```diff
@@ -53,3 +68,63 @@ def download(extension):
         download_btn("Download as {}".format(extensions_mapping[extension]))
     except:
         raise ValueError("Unknown extention. check the extentions_mapping")
+
+
+def add_deployment(filename, runtime_name=None, enable_deploy=True, cancel=False):
```

Comment:
cfme.web_ui.mixins' name is picked wrong, mixin should be a class :)

So, if I understand it correctly, the add_deployment is specific for Server, is that right? Then why not make a mixin class without add_deployment and create a separate mixin class for the add_deployment specifically. Then the Server would inherit both ...

How does that sound?